azext_iot/operations/dps.py

```diff
@@ -558,8 +558,38 @@ def iot_dps_device_enrollment_group_delete(
 
 
 def iot_dps_compute_device_key(
-    cmd, symmetric_key, registration_id,
+    cmd,
+    registration_id,
+    client=None,
+    enrollment_id=None,
+    dps_name=None,
+    resource_group_name=None,
+    symmetric_key=None,
 ):
+    if symmetric_key is None:
+        if not all([client, dps_name, resource_group_name, enrollment_id]):
+            raise CLIError(
+                "Please provide DPS enrollment group idenfitiers (Device Provisioning Service name via "
+                "--dps-name, Enrollment ID via --enrollment-id, and resource group via --resource-group "
+                "or -g) or the enrollment group symmetric key via --symmetric-key or --key."
+            )
+
+        target = get_iot_dps_connection_string(cmd, client, dps_name, resource_group_name)
+        try:
+            resolver = SdkResolver(target=target)
+            sdk = resolver.get_sdk(SdkType.dps_sdk)
+            attestation = sdk.get_enrollment_group_attestation_mechanism(
+                enrollment_id, raw=True
+            ).response.json()
+            if attestation.get("symmetricKey") is None:
```

Comment:
I think evaluating the attestation record type property to ensure it is of type symmetricKey would be better for this check. We also wouldn't need to hardcode 'x509' or a future attestation type.

Reply:
Updated the conditional and error message - there is also tpm as a type.